enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,391 @@
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Operator SDK currently ships the Ansible and Helm Operator base images
+downstream. The purpose of this enhancement is to add a supported Operator SDK
+base image as well as an operator-sdk binary. In addition, to add more
+automation to syncing upstream with downstream.
+
+## Motivation
+
+Deliver a properly supported Operator SDK downstream to support OpenShift
+developers.
+
+### Goals
+
+The primary goal is to deliver a new base image with the Operator SDK that
+can be used by operator developers. Deliver an operator-sdk binary similar to
+how oc is delivered to customers. Continue to deliver the Ansible and Helm
+operator base image.
+
+Another goal is to automate the syncing of upstream to downstream to make it
+easier to bring in upstream releases.
+
+### Non-Goals
+
+We will not deliver an RPM based operator-sdk package as part of this. SDK
+downstream scaffolding files to use downstream images this will be done when
+Phase 2 plugins are done. For now folks will have to modify their operators to
+pull from downstream images.
+
+## Proposal
+
+### User Stories [optional]
+
+* Deliver an operator-sdk binary
+  * As an operator developer, I would want the SDK that I use to support the supported OpenShift clusters.
+  * As a Red Hat operator customer, I would want a supported operator-sdk binary.
+* Deliver an operator-sdk base image
+  * As an operator developer, I would want a supported Operator SDK base image to use to build my own operators.
+  * As a Red Hat operator customer, I would want to have a released downstream Operator SDK base image to use to build my own operators.
+  * Provide operator-lib downstream
+* Create shell-script to help automate the process
+  * As an operator-sdk developer, I would like a way to easily bring upstream SDK releases downstream for release with OpenShift
+* Continue delivering Ansible and Helm operator base images
+  * As an Ansible operator developer, I would want to continue having a released downstream Ansible operator image I could base my operator on.
+  * As a Helm operator developer, I would want to continue having  a released downstream Helm operator image I could base my operator on.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Repos
+
+Mirror the upstream Operator SDK repos downstream in the openshift GitHub
+organization. We will have a one to one upstream to downstream repo.
+
+* operator-sdk - github.com/openshift/operator-sdk
+* operator-lib - github.com/openshift/operator-lib
+* ocp-release-operator-sdk - to be retired after the above repos are working
+
+The downstream repo will contain a special _overlay_ branch that contains all of
```

Comment:
So this means -  `_overlay_ ` branch would not contain any fixes or changes in mainline code. Downstream/upstream release branches will be in sync.

Reply:
@varshaprasad96 I think we will keep patches in the `patches` directory of the  _overlay_ branch as `.patch` files. The idea is to limit the conflicts that arise from updating the `release-x.y` branches. So the fix for downstream test would likely be in that `patches` directory. This might not be tenable to keep doing patches, if that becomes the case we might have to switch to something different keeping in mind conflicts that might arise during the sync.